apps/survey/app/src/script.js

```diff
@@ -0,0 +1,154 @@
+import Aragon from '@aragon/client'
+import { combineLatest } from './rxjs'
+import surveySettings, { hasLoadedSurveySettings } from './survey-settings'
+
+const app = new Aragon()
+
+// Hook up the script as an aragon.js store
+app.store(async (state, { event, returnValues }) => {
+  let nextState = {
+    ...state,
+    // Fetch the app's settings, if we haven't already
+    ...(!hasLoadedSurveySettings(state) ? await loadSurveySettings() : {}),
+  }
+
+  switch (event) {
+    case 'StartSurvey':
+      nextState = await startSurvey(nextState, returnValues)
+      break
+    case 'CastVote':
+      nextState = await castVote(nextState, returnValues)
+      break
+    default:
+      break
+  }
+
+  return nextState
+})
+
+/***********************
+ *                     *
+ *   Event Handlers    *
+ *                     *
+ ***********************/
+
+async function startSurvey(state, { surveyId }) {
+  const transform = ({ data, ...survey }) => ({
+    ...survey,
+    data: { ...data },
```

Comment:
A _slightly_ better way would be to pass the entire object through (avoids a new object if the transpiler / compiler isn't smart and is cleaner in intent); I've been thinking this is probably what the `transform` argument should default to anyway.

It'd look like:

```js
const transform = survey => survey
```

[See an example in the voting app's script](https://github.com/aragon/aragon-apps/blob/master/apps/voting/app/src/script.js#L58).